### ISSUE TITLE
Rule proposal: no-implicit-any

### ISSUE THREAD

**Issue by calvinlee2022** [typescript-eslint/typescript-eslint#5200]
### Before You File a Proposal Please Confirm You Have Done The Following...

- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3A%22enhancement%3A+new+plugin+rule%22) and found none that match my proposal.
- [X] I have searched the [current rule list](https://typescript-eslint.io/rules/#supported-rules) and found no rules that match my proposal.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### My proposal is suitable for this project

- [X] My proposal specifically checks TypeScript syntax, or it proposes a check that requires type information to be accurate.
- [X] My proposal is not a "formatting rule"; meaning it does not just enforce how code is formatted (whitespace, brace placement, etc).
- [X] I believe my proposal would be useful to the broader TypeScript community (meaning it is not a niche proposal).

### Description

I want to have a strict checking in code。

### Fail Cases

```typescript
function(a){
}
```

### Pass Cases

```typescript
function(a: string){

}
```

**Comment by bradzacher** [comment]
TypeScript had this built in as a compiler option.
We do not duplicate checks TS has.
